invenio_app_rdm/records_ui/views/records.py

```diff
@@ -187,11 +187,27 @@ def record_detail(
                 if p == "doi" and "datacite" in v
             ]
             if datacite_provider:
-                datacite_provider = datacite_provider[0]
-                parent_doi = datacite_provider.client.generate_doi(
-                    record._record.parent
+                should_mint_parent_doi = True
+                is_doi_required = (
+                    current_app.config.get("RDM_PARENT_PERSISTENT_IDENTIFIERS", {})
+                    .get("doi", {})
+                    .get("required")
                 )
-                record_ui["ui"]["new_draft_parent_doi"] = parent_doi
+                if not is_doi_required:
+                    # check if the draft has a reserved doi and mint parent doi only in that case
+                    record_doi = record._record.pids.get("doi", {})
+                    is_doi_reserved = record_doi.get(
+                        "provider", ""
+                    ) == "datacite" and record_doi.get("identifier")
+                    if not is_doi_reserved:
+                        should_mint_parent_doi = False
+
+                if should_mint_parent_doi:
+                    datacite_provider = datacite_provider[0]
+                    parent_doi = datacite_provider.client.generate_doi(
```

Comment:
Question: I guess that the users can still change their mind after generating the parent DOI and remove the DOI, resulting in a "orphan" parent DOI? (jsut asking to confirm if I understand this correctly) If that's the case I find it a bit weird, maybe something to fix in another issue, unless I am missing something.